aait/backend/group-2B/Persistence/Repositories/InteractionRepository.cs

```diff
@@ -1,15 +1,46 @@
-using SocialSync.Application.Contracts.Persistence;
+using Microsoft.EntityFrameworkCore;
 using SocialSync.Domain.Entities;
+using SocialSync.Application.Contracts.Persistence;
+
 
 namespace SocialSync.Persistence.Repositories;
 
 public class InteractionRepository : GenericRepository<Interaction>, IInteractionRepository
 {
-    private readonly SocialSyncDbContext _dbContext;
+    private readonly SocialSyncDbContext _context;
+
+    public InteractionRepository(SocialSyncDbContext context)
+        : base(context)
+    {
+        _context = context;
+    }
 
-    public InteractionRepository(SocialSyncDbContext dbContext)
-        : base(dbContext)
+    public async Task<List<Interaction>> GetAllCommentInteractionsAsync(int postId)
     {
-        _dbContext = dbContext;
+        var allComments = await _context.Interactions
+            .Where(x => x.PostId == postId && x.Type == InteractionType.Comment)
+            .ToListAsync();
+        return allComments;
+    }
+
+    public async Task<Interaction> LikeUnlikeInteractionAsync(Interaction interaction)
```

Comment:
what is the purpose of this method and its corresponse handler(It should do one thing (deleting or adding)

Reply:
The method is used because we have the same api that is used for liking and disliking we assumed it would be like on social medias like instagram.